### ISSUE TITLE
[bug] cannot manage rules with funny names

### ISSUE THREAD

**Issue by GeoffWilliams** [GeoffWilliams/puppet-windows_firewall#3]
eg
```
Notice: /Stage[main]/Main/Windows_firewall_rule[@{microsoft.windows.shellexperiencehost_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://microsoft.windows.shellexperiencehost/resources/pkgdisplayname}]/ensure: removed
Notice: (windows_firewall) deleting rule '@{microsoft.windows.shellexperiencehost_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://microsoft.windows.shellexperiencehost/resources/pkgdisplayname}'
Error: /Stage[main]/Main/Windows_firewall_rule[@{microsoft.windows.shellexperiencehost_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://microsoft.windows.shellexperiencehost/resources/pkgdisplayname}]: Could not evaluate: Execution of 'C:/Windows/system32/netsh.exe advfirewall firewall delete rule name="@{microsoft.windows.shellexperiencehost_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://microsoft.windows.shellexperiencehost/resources/pkgdisplayname}"' returned 1: No rules match the specified criteria.
```

**Comment by GeoffWilliams** [comment]
Fixing this requires breaking changes, to align with powershell, eg:
`profiles` -> `profile`
`localip` -> `local_address`
`remoteip` -> `remote_address`
`localport` -> `local_port`
`remoteport` -> `remote_port`
`edgetraversal` -> `edge_traversal_policy`
`interfacetypes` -> `interface_type`
`security` -> gone
`rule_source` -> gone


Some values need to change to:
`in` -> `inbound`
`out` -> `outbound`

**Comment by GeoffWilliams** [comment]
https://gist.github.com/GeoffWilliams/8c0412057b2db6ea629de677311c2c20

**Comment by GeoffWilliams** [comment]
Full list - note dups:
```
Rule Name:                            @{Microsoft.AAD.BrokerPlugin_1000.17134.1.0_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.AAD.BrokerPlugin/resources/PackageDisplayName}
Rule Name:                            @{Microsoft.AAD.BrokerPlugin_1000.17134.1.0_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.AAD.BrokerPlugin/resources/PackageDisplayName}
Rule Name:                            @{Microsoft.AccountsControl_10.0.17134.1_neutral__cw5n1h2txyewy?ms-resource://Microsoft.AccountsControl/Resources/DisplayName}
Rule Name:                            @{Microsoft.BingWeather_4.22.3254.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.BingWeather/Resources/ApplicationTitleWithBranding}
Rule Name:                            @{Microsoft.DesktopAppInstaller_1.0.12271.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.DesktopAppInstaller/Resources/appDisplayName}
Rule Name:                            @{Microsoft.DesktopAppInstaller_1.0.12271.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.DesktopAppInstaller/Resources/appDisplayName}
Rule Name:                            @{Microsoft.GetHelp_10.1706.10441.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.GetHelp/Resources/appDisplayName}
Rule Name:                            @{Microsoft.Getstarted_6.9.10602.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.Getstarted/Resources/AppStoreName}
Rule Name:                            @{Microsoft.LockApp_10.0.17134.1_neutral__cw5n1h2txyewy?ms-resource://Microsoft.LockApp/resources/AppDisplayName}
Rule Name:                            @{Microsoft.Messaging_3.38.22001.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.Messaging/Resources/AppStoreName}
Rule Name:                            @{Microsoft.Messaging_3.38.22001.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.Messaging/Resources/AppStoreName}
Rule Name:                            @{Microsoft.Microsoft3DViewer_2.1803.8022.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.Microsoft3DViewer/Common.View.UWP/Resources/StoreAppName}
Rule Name:                            @{Microsoft.MicrosoftEdge_42.17134.1.0_neutral__8wekyb3d8bbwe?ms-resource://Microsoft.MicrosoftEdge/Resources/AppName}
Rule Name:                            @{Microsoft.MicrosoftEdge_42.17134.1.0_neutral__8wekyb3d8bbwe?ms-resource://Microsoft.MicrosoftEdge/Resources/AppName}
Rule Name:                            @{Microsoft.MicrosoftOfficeHub_17.8918.5926.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.MicrosoftOfficeHub/officehubintl/AppManifest_GetOffice_DisplayName}
Rule Name:                            @{Microsoft.MicrosoftOfficeHub_17.8918.5926.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.MicrosoftOfficeHub/officehubintl/AppManifest_GetOffice_DisplayName}
Rule Name:                            @{Microsoft.MSPaint_3.1803.5027.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.MSPaint/resources/AppName}
Rule Name:                            @{Microsoft.OneConnect_4.1801.521.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.OneConnect/OneConnect/AppStoreName}
Rule Name:                            @{Microsoft.OneConnect_4.1801.521.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.OneConnect/OneConnect/AppStoreName}
Rule Name:                            @{Microsoft.People_10.3.10452.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.People/Resources/AppStoreName}
Rule Name:                            @{Microsoft.PPIProjection_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.PPIProjection/resources/ProductName}
Rule Name:                            @{Microsoft.PPIProjection_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.PPIProjection/resources/ProductName}
Rule Name:                            @{Microsoft.PPIProjection_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.PPIProjection/resources/ProductName}
Rule Name:                            @{Microsoft.PPIProjection_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.PPIProjection/resources/ProductName}
Rule Name:                            @{Microsoft.SkypeApp_12.13.274.0_x64__kzf8qxf38zg5c?ms-resource://Microsoft.SkypeApp/Resources/SkypeVideo_ProductName}
Rule Name:                            @{Microsoft.SkypeApp_12.13.274.0_x64__kzf8qxf38zg5c?ms-resource://Microsoft.SkypeApp/Resources/SkypeVideo_ProductName}
Rule Name:                            @{Microsoft.StorePurchaseApp_11712.1801.10002.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.StorePurchaseApp/Resources/DisplayTitle}
Rule Name:                            @{Microsoft.Todos_1.41.12842.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.Todos/Resources/app_name_ms_todo}
Rule Name:                            @{Microsoft.Todos_1.41.12842.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.Todos/Resources/app_name_ms_todo}
Rule Name:                            @{Microsoft.Windows.Apprep.ChxApp_1000.17134.1.0_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.Apprep.ChxApp/resources/DisplayName}
Rule Name:                            @{Microsoft.WindowsCalculator_10.1712.10601.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.WindowsCalculator/Resources/AppStoreName}
Rule Name:                            @{Microsoft.WindowsCamera_2017.1117.80.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.WindowsCamera/Resources/AppStoreName}
Rule Name:                            @{Microsoft.Windows.CloudExperienceHost_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.CloudExperienceHost/resources/appDescription}
Rule Name:                            @{Microsoft.Windows.CloudExperienceHost_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.CloudExperienceHost/resources/appDescription}
Rule Name:                            @{Microsoft.Windows.CloudExperienceHost_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.CloudExperienceHost/resources/appDescription}
Rule Name:                            @{Microsoft.Windows.CloudExperienceHost_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.CloudExperienceHost/resources/appDescription}
Rule Name:                            @{microsoft.windowscommunicationsapps_17.8827.22055.0_x64__8wekyb3d8bbwe?ms-resource://microsoft.windowscommunicationsapps/hxoutlookintl/AppManifest_OutlookDesktop_DisplayName}
Rule Name:                            @{microsoft.windowscommunicationsapps_17.8827.22055.0_x64__8wekyb3d8bbwe?ms-resource://microsoft.windowscommunicationsapps/hxoutlookintl/AppManifest_OutlookDesktop_DisplayName}
Rule Name:                            @{Microsoft.Windows.ContentDeliveryManager_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.ContentDeliveryManager/resources/AppDisplayName}
Rule Name:                            @{Microsoft.Windows.ContentDeliveryManager_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.ContentDeliveryManager/resources/AppDisplayName}
Rule Name:                            @{Microsoft.Windows.Cortana_1.10.7.17134_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.Cortana/resources/PackageDisplayName}
Rule Name:                            @{Microsoft.Windows.Cortana_1.10.7.17134_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.Cortana/resources/PackageDisplayName}
Rule Name:                            @{Microsoft.WindowsFeedbackHub_1.1712.612.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.WindowsFeedbackHub/Resources/AppStoreName}
Rule Name:                            @{Microsoft.WindowsFeedbackHub_1.1712.612.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.WindowsFeedbackHub/Resources/AppStoreName}
Rule Name:                            @{Microsoft.Windows.HolographicFirstRun_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.HolographicFirstRun/resources/PkgDisplayName}
Rule Name:                            @{Microsoft.WindowsMaps_5.1711.10401.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.WindowsMaps/Resources/AppStoreName}
Rule Name:                            @{Microsoft.Windows.OOBENetworkCaptivePortal_10.0.17134.1_neutral__cw5n1h2txyewy?ms-resource://Microsoft.Windows.OOBENetworkCaptivePortal/Resources/AppDisplayName}
Rule Name:                            @{Microsoft.Windows.ParentalControls_1000.17134.1.0_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.ParentalControls/resources/DisplayName}
Rule Name:                            @{Microsoft.Windows.PeopleExperienceHost_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.PeopleExperienceHost/resources/PkgDisplayName}
Rule Name:                            @{Microsoft.Windows.Photos_2018.18011.15918.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.Windows.Photos/Resources/AppStoreName}
Rule Name:                            @{Microsoft.Windows.Photos_2018.18011.15918.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.Windows.Photos/Resources/AppStoreName}
Rule Name:                            @{Microsoft.Windows.SecHealthUI_10.0.17134.1_neutral__cw5n1h2txyewy?ms-resource://Microsoft.Windows.SecHealthUI/resources/PackageDisplayName}
Rule Name:                            @{Microsoft.Windows.SecureAssessmentBrowser_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.SecureAssessmentBrowser/Resources/PackageDisplayName}
Rule Name:                            @{Microsoft.Windows.ShellExperienceHost_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.ShellExperienceHost/resources/PkgDisplayName}
Rule Name:                            @{Microsoft.Windows.ShellExperienceHost_10.0.17134.1_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.Windows.ShellExperienceHost/resources/PkgDisplayName}
Rule Name:                            @{Microsoft.WindowsStore_11712.1001.23.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.WindowsStore/Resources/StoreTitle}
Rule Name:                            @{Microsoft.WindowsStore_11712.1001.23.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.WindowsStore/Resources/StoreTitle}
Rule Name:                            @{Microsoft.XboxGameCallableUI_1000.17134.1.0_neutral_neutral_cw5n1h2txyewy?ms-resource://Microsoft.XboxGameCallableUI/resources/PkgDisplayName}
Rule Name:                            @{Microsoft.XboxIdentityProvider_12.36.15002.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.XboxIdentityProvider/Resources/DisplayName}
Rule Name:                            @{Microsoft.ZuneMusic_10.17112.19011.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.ZuneMusic/resources/IDS_MANIFEST_MUSIC_APP_NAME}
Rule Name:                            @{Microsoft.ZuneMusic_10.17112.19011.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.ZuneMusic/resources/IDS_MANIFEST_MUSIC_APP_NAME}
Rule Name:                            @{Microsoft.ZuneVideo_10.17112.19011.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.ZuneVideo/resources/IDS_MANIFEST_VIDEO_APP_NAME}
Rule Name:                            @{Microsoft.ZuneVideo_10.17112.19011.0_x64__8wekyb3d8bbwe?ms-resource://Microsoft.ZuneVideo/resources/IDS_MANIFEST_VIDEO_APP_NAME}
```

**Comment by GeoffWilliams** [comment]
fixed in 0.2.0
